pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -45,11 +45,13 @@ func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.H
 	}
 	handler = genericapifilters.WithAuthentication(handler, c.RequestContextMapper, insecureSuperuser{}, nil)
 	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
-	handler = genericfilters.WithPanicRecovery(handler)
 	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.RequestContextMapper, c.LongRunningFunc, c.RequestTimeout)
 	handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.RequestContextMapper, c.LongRunningFunc)
+	handler = genericfilters.WithWaitGroup(handler, c.RequestContextMapper, c.LongRunningFunc, c.HandlerChainWaitGroup)
 	handler = genericapifilters.WithRequestInfo(handler, server.NewRequestInfoResolver(c), c.RequestContextMapper)
 	handler = apirequest.WithRequestContext(handler, c.RequestContextMapper)
+	// PanicRecovery handler must registered last and execute first in the chain.
```

Comment:
as we don't give reasons for the order of the other fliters, I would drop this comment.

Reply:
Yes, but filter PanicRecovery handler placed last, execute first is appropriate

Reply:
Why?

Reply:
Panic handler should be executed first, otherwise if other handler panic no recovery.

Reply:
I guess @sttts means we should drop the comment.

The change in code is OK to him.

Reply:
ok, I know.